Sources/StreamVideo/WebRTC/v2/PeerConnection/MediaAdapters/LocalMediaAdapters/LocalScreenShareMediaAdapter.swift

```diff
@@ -517,4 +517,20 @@ final class LocalScreenShareMediaAdapter: LocalMediaAdapting, @unchecked Sendabl
             subsystems: .webRTC
         )
     }
+
+    /// We only want to register the primaryTrack only when we are about to publish for the first time.
+    private func registerPrimaryTrackIfPossible() {
+        guard !hasRegisteredPrimaryTrack else {
+            return
+        }
+
+        subject.send(
```

Comment:
can this process fail and we end up not registering the primary track?

Reply:
The only thing it does is to send it to a subject, so no errors here.